x-pack/plugin/core/src/main/java/org/elasticsearch/license/License.java

```diff
@@ -780,22 +780,4 @@ public Builder validate() {
         }
     }
 
-    /**
-     * Returns <code>true</code> iff the license is a production licnese
-     */
-    public boolean isProductionLicense() {
```

Comment:
I replaced this with an explicit "is TLS required" method in XPackLicenseState.

I droppped the "production license" terminology, because it was a bit confusing (which is this gap has only just been identified).
Why wasn't basic a production license before? Should it be one? It's perfectly fine to run basic in production, even though it doesn't have paid support.

I moved it from License to `XPackLicenseState` because
1. All the "is XXX allowed" methods are there, so that seemed more reasonable from an appropriate compartmentalisation problem (and helps with discoverability of the code)
2. To do the "is TLS required" method well it needs to look at whether security is explicitly enabled, for which all the logic is in XPackLicenseState.